src/ocrd/decorators/__init__.py

```diff
@@ -71,6 +71,8 @@ def ocrd_cli_wrap_processor(
     initLogging()
 
     LOG = getLogger('ocrd.cli_wrap_processor')
+    assert kwargs['input_file_grp'] is not None
+    assert kwargs['output_file_grp'] is not None
```

Comment:
What about processors without output file group parameters (e.g., `ocrd-segment-evaluate`)? Is the cardinality in the `setup()` method just set to 0?

Reply:
At the moment we do not have any actual processors without output file group. Even the `*-evaluate` processors produce some kind of JSON and add that to the METS. And since I [proposed](https://github.com/OCR-D/spec/issues/172) having a separate CLI spec for **evaluators**, I believe we should change the spec for **processors** to require exactly one output file grp.

In `setup` implementations, developers are free to make cardinality checks as they please. But the superclass needs to have `Processor.output_file_grp` well-defined so the default implementation of `process_page_file` can do something useful.

Reply:
> At the moment we do not have any actual processors without output file group.

Oh, in fact, there is: ocrd-docstruct. It only writes METS structMap divs. But perhaps we could allow using an **empty output fileGrp** for these cases. (The default implementation process_page_file will have to be overwritten anyway.)

Reply:
In case of docstruct, even `process_workspace` will have to be customised (no input files to iterate over).

An empty string (still not `None`) for such cases would be fine with me.

If we start differentiating construction from calling context, there will be no more need for these assertions – we could simply pass them as mandatory type-checked args to `process_workspace`.

Reply:
> In case of docstruct, even `process_workspace` will have to be customised (no input files to iterate over).

OK, but considering that docstruct is actually working workspace-wide, I think we can live with that.

> An empty string (still not `None`) for such cases would be fine with me.

:+1:

> 
> If we start differentiating construction from calling context, there will be no more need for these assertions – we could simply pass them as mandatory type-checked args to `process_workspace`.

Also :+1: